### PR TITLE
Allow multiple channels to be passed to Client.Join

### LINKED ISSUE #82: JoinMany func
would be nice to add a .JoinMany(channels ...string) func 
cram as many channels as possible in 512 bytes and send them each to the server

kinda bad calling join on each channel when you have a few hundred channels

### PR DESCRIPTION
**Pull request by brattonross**
Implements the change suggested by @IAmFletcher in #82.

Changes the Client.Join signature from
```go
func (c *Client) Join(channel string)
```
to
```go
func (c *Client) Join(channels ...string) []string
```

Fixes #82

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -517,22 +517,66 @@ func (c *Client) Whisper(username, text string) {
...
+		c.channelUserlistMutex.Lock()
+		c.channelUserlist[channel] = map[string]bool{}
+		c.channelUserlistMutex.Unlock()
+	}
 	c.channelsMtx.Unlock()
+
+	return remainder
+}
+
+var maxMessageLength = 510
```
Comment (gempir) [inline client.go]: please move this to the IRC file and make a constant out of it. It's not variable.

File: client.go [hunk 2]
```diff
@@ -517,22 +517,66 @@ func (c *Client) Whisper(username, text string) {
...
 
-// Join enter a twitch channel to read more messages
-func (c *Client) Join(channel string) {
-	channel = strings.ToLower(channel)
-
-	// If we don't have the channel in our map AND we have an
-	// active connection, explicitly join before we add it to our map
+// Join enter a twitch channel to read more messages.
+// Any channels which could not be joined due to the maximum
+// message length being reached are returned in a slice.
```
Comment (gempir) [inline client.go]: Our client should handle that for the user. The user should not care what the message limits of IRC are or TMI are, we abstract that. 

So no return value please.

**Comment by gempir** [comment]
Thanks for your contribution, I still have some issues with it, but it looks promising. Tests look good.

**Comment by brattonross** [comment]
Thanks for your feedback @gempir, I've updated the implementation with your requested changes

**Comment by gempir** [comment]
Since you use the new `strings.Builder` from go 1.10 can you update the travis file to remove older versions of go?

File: client.go (hunk 1 above)
Comment (brattonross) [reply client.go]: Moved to IRC file and made constant

**Comment by gempir** [comment]
This looks good to me. Thanks for your contribution @brattonross

**Pull request merged**